### ISSUE TITLE
Confusing error messages when user doesn't specify `version`

### ISSUE THREAD

**Issue by kidonng** [pnpm/action-setup#54]
It seems this action _requires_ a `package.json` file to be present:

https://github.com/pnpm/action-setup/blob/6e1964dde3397a825e79e4607ad57f3f7ca2c7cb/src/install-pnpm/run.ts#L13-L14

Even if users only use `pnpx` and do not install packages locally, they have to provide a `package.json`, which is inconvenient.

Will the action support using it without a `package.json`?

**Comment by KSXGitHub** [comment]
You misinterpreted the code. Have you tried running this action without `package.json`?

**Comment by kidonng** [comment]
Yep: https://github.com/kidonng/api/actions/runs/3051844927/jobs/4920548325#step:4:7

**Comment by kidonng** [comment]
Ah, I think I got what's happening. I didn't specify a `version`, so the action is finding the `packageManager` field in `package.json`, but I don't have one.

Sorry, my assumption is wrong 🙈

**Comment by KSXGitHub** [comment]
I think the error messages can be improved.
